### ISSUE TITLE
Microsoft.OneDriveSDK and Microsoft.OneDriveSDKAuthentication don't work with Project Reunion desktop running on Win32

### ISSUE THREAD

**Issue by gcaughey** [microsoft/microsoft-ui-xaml#4526]
**Describe the bug**

Although Microsoft.OneDriveSDK and Microsoft.OneDriveSDKAuthentication nuget packages ca be used successfully with Project Reunion 0.5 apps running on UWP, the same code fails running on Win32 with a strange error message on compile: Type universe cannot resolve assembly: System.Windows.Forms, Version=4.0.0.0,

**Steps to reproduce the bug**

1. Create new project using Blank Project Reunion desktop template
2. Add nuget references to Microsoft.OneDriveSDK and Microsoft.OneDriveSDKAuthentication
3. Add any code that references MsaAuthenticationProvider 
4. Attempt to compile app. Compiler error references Windows.Forms for some reason!

**Expected behavior**

App should compile.

**Screenshots**

**Version Info**

NuGet package version: 

Microsoft.ProjectReunion 0.5.0-prerelease

| Windows 10 version                  | Saw the problem? |
| :--------------------------------- | :-------------------- |
| Insider Build (21332)              | Yes  |
| May 2020 Update (19041)            |    |
| November 2019 Update (18363)       |    |
| May 2019 Update (18362)            |    |
| October 2018 Update (17763)        |    |

**Additional context**

This code works fine on Project Reunion 0.5 app running on UWP. It is only an issue for Project Reunion desktop app when running on Win32.

**Comment by gcaughey** [comment]
Is there anything I should try to pin this down? It seems so odd it's only an issue running Project Reunion 0.5 desktop on Win32 and not running Project Reunion 0.5 on UWP.

**Comment by riverar** [comment]
**tl;dr:** Don't think this is a supported scenario in WinUI 3 for desktop.

Warnings and errors after following repro steps:
```
warning NU1701: Package 'Microsoft.Graph.Core 1.3.1' was restored using '.NETFramework,Version=v4.6.1, .NETFramework,Version=v4.6.2, .NETFramework,Version=v4.7, .NETFramework,Version=v4.7.1, .NETFramework,Version=v4.7.2, .NETFramework,Version=v4.8, .NETFramework,Version=v4.6.1' instead of the project target framework 'net5.0-windows10.0.19041'. This package may not be fully compatible with your project.
warning NU1701: Package 'Microsoft.OneDriveSDK 2.0.7' was restored using '.NETFramework,Version=v4.6.1, .NETFramework,Version=v4.6.2, .NETFramework,Version=v4.7, .NETFramework,Version=v4.7.1, .NETFramework,Version=v4.7.2, .NETFramework,Version=v4.8, .NETFramework,Version=v4.6.1' instead of the project target framework 'net5.0-windows10.0.19041'. This package may not be fully compatible with your project.
warning NU1701: Package 'Microsoft.OneDriveSDK.Authentication 1.0.10' was restored using '.NETFramework,Version=v4.6.1, .NETFramework,Version=v4.6.2, .NETFramework,Version=v4.7, .NETFramework,Version=v4.7.1, .NETFramework,Version=v4.7.2, .NETFramework,Version=v4.8, .NETFramework,Version=v4.6.1' instead of the project target framework 'net5.0-windows10.0.19041'. This package may not be fully compatible with your project.
XamlCompiler error WMC1006: Cannot resolve Assembly or Windows Metadata file 'Type universe cannot resolve assembly: System.Windows.Forms, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089.'
```

My initial analysis:

`Microsoft.OneDriveSdk.Authentication` ships separate libraries for UWP and Desktop scenarios. So in your UWP project, Microsoft strips away some incompatible desktop capabilities and Things Work Great™. In the desktop scenario, [some dialog capabilities remain](https://github.com/OneDrive/onedrive-sdk-dotnet-msa-auth-adapter/blob/a7b4696e7a0451caa6ff4b7c2c39628846c45c5a/src/OneDrive.Sdk.Authentication.Desktop/Web/FormsWebDialog.cs). 

So to support `System.Windows.Forms` usage, normally you'd start by adding `<UseWindowsForms>true</UseWindowsForms>` and `<UseWpf>true</UseWpf>` to your project. But this isn't supported in WinUI 3 for desktop. (I believe this creates xaml ambiguities that haven't been resolved yet.)

**Comment by gcaughey** [comment]
Makes sense. I'll kick the ball along to the OneDrive folks. Thanks.

**Comment by gcaughey** [comment]
Here's the matching issue on OneDrive issues: [https://github.com/OneDrive/onedrive-sdk-dotnet-msa-auth-adapter/issues/71](url)

**Comment by riverar** [comment]
@gcaughey Not sure what you're using to compose replies but both of your links are pointing to a black hole :smile:

**Comment by shaheedmalik** [comment]
> @gcaughey Not sure what you're using to compose replies but both of your links are pointing to a black hole 😄

Probably
https://github.com/OneDrive/onedrive-sdk-dotnet-msa-auth-adapter/issues/70

https://github.com/OneDrive/onedrive-sdk-dotnet-msa-auth-adapter/issues/71

**Comment by bpulliam** [comment]
Closing stale external bug. If you are still seeing this issue, please open a new issue and include updated repro steps and related logs/diagnostics.
